aiida_vasp/calcs/vasp.py

```diff
@@ -303,8 +303,10 @@ def write_poscar(self, dst):  # pylint: disable=unused-argument
         settings = self.inputs.get('settings')
         settings = settings.get_dict() if settings else {}
         poscar_precision = settings.get('poscar_precision', 10)
-        positions_dof = self.inputs.get('dynamics', {}).get('positions_dof')
-        if positions_dof is not None:
+        dynamics = self.inputs.get('dynamics')
+        if dynamics is not None:
+            dynamics = dynamics.get_dict() if dynamics else {}
```

Comment:
Do we need the `if else`? We are already inside is not `None`.

Reply:
You are right, I was following how it was done above in the settings, but it is probably not needed at all. I'll double check with my tests.